### PR TITLE
Image cropping

### PR DESCRIPTION
**Pull request by Rodrigmi**
Added the image cropping file to prep our images for running them through our CNN.

### CODE REVIEW

**Review by julianweisbord — APPROVED** [review]
We'll have to go back and add PEP8 to this file soon but for now it looks good.

**Comment by julianweisbord** [comment]
@mccallm1 can you review this PR?

**Review by mccallm1 — COMMENTED** [review]
Should probably make vars like object class non-hard coded, command line args for example

**Comment by julianweisbord** [comment]
:+1: Or both via default args if no command line args are provided

**Review by julianweisbord — CHANGES_REQUESTED** [review]
PEP8, comments

**Comment by julianweisbord** [comment]
I found this module to be difficult to use. The directory structure is too specific, and instead of creating directories to save the newly cropped images, it requires that you already have them made. More comments would also be helpful, especially regarding how the user can change the file to make the cropped images larger or smaller.

File: src/crop_image.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
```
Comment (julianweisbord) [inline src/crop_image.py]: Add a comment header, see https://github.com/julianweisbord/Deep-Learning-For-Object-Recognition-on-a-Mobile-Robot/blob/master/src/learning/model.py

File: src/crop_image.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+from os.path import isfile, join, splitext
+from PIL import Image
+import numpy as np
+import cv2
+import glob
+
+
+def crop(image_path, coords, saved_location):
+    """
+    @param image_path: The path to the image to edit
```
Comment (julianweisbord) [inline src/crop_image.py]: Its best if function headers should have a description, the parameters, and the return value, see https://github.com/julianweisbord/Deep-Learning-For-Object-Recognition-on-a-Mobile-Robot/blob/master/src/learning/model.py

File: src/crop_image.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    # Arguments
+        None
+    # Returns
+        Output: None
+'''
+
+
+def main():
+
+    for instance_num in range(5):
```
Comment (julianweisbord) [inline src/crop_image.py]: No magic numbers

File: src/crop_image.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    @param image_path: The path to the image to edit
+    @param coords: A tuple of x/y coordinates (x1, y1, x2, y2)
+    @param saved_location: Path to save the cropped image
+    """
+    image_obj = Image.open(image_path)
+    cropped_image = image_obj.crop(coords)
+    cropped_image.save(saved_location)
+
+
+'''
```
Comment (julianweisbord) [inline src/crop_image.py]: This should be inside main and formatted like the other function headers

File: src/crop_image.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        files = [f for f in listdir(text_file_path) if f.endswith(".txt")]
+        for file in files:
+
+            # Break if we've reached the number of images in directory
+            if index == 80:
+                break
+
+            file_base = splitext(file)[0]
+            image_file_name = file_base + ".png"
+            print("Index: ", index)  # Printing index of current current image
```
Comment (julianweisbord) [inline src/crop_image.py]: Make sure that comments have correct grammar and are full sentences

**Review by mccallm1 — CHANGES_REQUESTED** [review]
Function headers are probably the most important addition based on Ben's feedback

**Comment by Rodrigmi** [comment]
The image cropping file now creates directories if they don' exist to save the cropped images. It also takes command line arguments now for a path to the data_captured folder and the image class that you want to crop.

File: src/crop_image.py [hunk 6]
```diff
@@ -1,69 +1,122 @@
-#!/usr/bin/env python
 
+'''
+Created on March 15th, 2018
+author: Michael Rodriguez
+sources: https://www.blog.pythonlibrary.org/2017/10/03/how-to-crop-a-photo-with-python/
+description: Crops images of a given object class to feed into neural network.
+'''
+# !/usr/bin/env python
 from os import listdir
```
Comment (julianweisbord) [inline src/crop_image.py]: I don't see the purpose importing a bunch of os functions, if you are still just going to import os

File: src/crop_image.py [hunk 7]
```diff
@@ -1,69 +1,122 @@
...
+import sys
+import argparse
 
 
 def crop(image_path, coords, saved_location):
     """
-    @param image_path: The path to the image to edit
-    @param coords: A tuple of x/y coordinates (x1, y1, x2, y2)
-    @param saved_location: Path to save the cropped image
+    crop - Image cropping function
```
Comment (julianweisbord) [inline src/crop_image.py]: Function headers https://github.com/julianweisbord/Deep-Learning-For-Object-Recognition-on-a-Mobile-Robot/blob/master/src/learning/model.py

File: src/crop_image.py [hunk 8]
```diff
@@ -1,69 +1,122 @@
...
+    The function takes an entire image class folder and loops through all
     instances within it cropping each one and saving it in a new directory.
     # Arguments
         None
     # Returns
         Output: None
-'''
+    '''
 
+    if (len(sys.argv) < 5):
```
Comment (julianweisbord) [inline src/crop_image.py]: No magic numbers

**Review by julianweisbord — DISMISSED** [review]

**Review by mccallm1 — DISMISSED** [review]

**Review by julianweisbord — APPROVED** [review]

**Pull request merged**